include/onnxruntime/core/graph/graph_nodes.h

```diff
@@ -132,18 +147,36 @@ class ValidNodes {
    private:
     TIterator current_;
     TIterator end_;
+    bool filter_;                        // store whether filter_func_ is not nullptr and contains a callable
```

Comment:
>filter_ [](start = 9, length = 7)

has_filter_?  filter_, filter_func_ and filter_node_fn_ make it very confusing

Reply:
Renamed to apply_filter_ and made the init `apply_filter_{filter_fn != nullptr}` to try and make it clearer